src/erf.jl

```diff
@@ -0,0 +1,143 @@
+const ra7  = -9.81432934416914548592e+00 # 0xC023A0EF, 0xC69AC25C 
+const sa1  =  1.96512716674392571292e+01 # 0x4033A6B9, 0xBD707687 
+const sa2  =  1.37657754143519042600e+02 # 0x4061350C, 0x526AE721 
+const sa3  =  4.34565877475229228821e+02 # 0x407B290D, 0xD58A1A71 
+const sa4  =  6.45387271733267880336e+02 # 0x40842B19, 0x21EC2868 
+const sa5  =  4.29008140027567833386e+02 # 0x407AD021, 0x57700314 
+const sa6  =  1.08635005541779435134e+02 # 0x405B28A3, 0xEE48AE2C 
+const sa7  =  6.57024977031928170135e+00 # 0x401A47EF, 0x8E484A93 
+const sa8  = -6.04244152148580987438e-02 # 0xBFAEEFF2, 0xEE749A62 
+
+# Coefficients for approximation to  erfc in [1/.35,28]
+const rb0  = -9.86494292470009928597e-03 # 0xBF843412, 0x39E86F4A
+const rb1  = -7.99283237680523006574e-01 # 0xBFE993BA, 0x70C285DE
+const rb2  = -1.77579549177547519889e+01 # 0xC031C209, 0x555F995A
+const rb3  = -1.60636384855821916062e+02 # 0xC064145D, 0x43C5ED98
+const rb4  = -6.37566443368389627722e+02 # 0xC083EC88, 0x1375F228
+const rb5  = -1.02509513161107724954e+03 # 0xC0900461, 0x6A2E5992
+const rb6  = -4.83519191608651397019e+02 # 0xC07E384E, 0x9BDC383F
+const sb1  =  3.03380607434824582924e+01 # 0x403E568B, 0x261D5190
+const sb2  =  3.25792512996573918826e+02 # 0x40745CAE, 0x221B9F0A
+const sb3  =  1.53672958608443695994e+03 # 0x409802EB, 0x189D5118
+const sb4  =  3.19985821950859553908e+03 # 0x40A8FFB7, 0x688C246A
+const sb5  =  2.55305040643316442583e+03 # 0x40A3F219, 0xCEDF3BE6
+const sb6  =  4.74528541206955367215e+02 # 0x407DA874, 0xE79FE763
+const sb7  = -2.24409524465858183362e+01 # 0xC03670E2, 0x42712D62
+
+function erfc1(x::Float64)
+    s = abs(x) - 1
+    P = @horner s pa0 pa1 pa2 pa3 pa4 pa5 pa6
+    Q = @horner s 1.0 qa1 qa2 qa3 qa4 qa5 qa6
+    return 1 - erx - P/Q
+end
+
+function erfc2(ix::UInt32, x::Float64)
+    if ix < 0x3ff40000  # |x| < 1.25
+        return erfc1(x)
+    end
+    x = abs(x) 
+    s = 1/(x*x)
+    if ix < 0x4006db6d # |x| < 1/.35 ~ 2.85714
+        R = @horner s ra0 ra1 ra2 ra3 ra4 ra5 ra6 ra7
+        S = @horner s 1.0 sa1 sa2 sa3 sa4 sa5 sa6 sa7 sa8
+    else # |x| > 1/.35
+        R = @horner s rb0 rb1 rb2 rb3 rb4 rb5 rb6
+        S = @horner s 1.0 sb1 sb2 sb3 sb4 sb5 sb6 sb7
+    end
+    z = x
+    z = set_low_word(z,UInt32(0))
+    return exp(-z*z-0.5625)*exp((z-x)*(z+x)+R/S)/x
+end
+
+function erf(x::Float64)
+    ix = get_high_word(x)
+    sign = Int(ix>>31)
+    ix &= 0x7fffffff
+    if ix >= 0x7ff00000 # erf(nan)=nan, erf(+-inf)=+-1
+        return 1-2*sign + 1/x
+    end
+    if ix < 0x3feb0000 # |x| < 0.84375
+        if ix < 0x3e300000 #|x| < 2**-28  avoid underflow
+            return 0.125*(8*x + efx8*x)
+        end
+        z = x*x
+        r = @horner z pp0 pp1 pp2 pp3 pp4
+        s = @horner z 1.0 qq1 qq2 qq3 qq4 qq5
+        y = r/s
+        return x + x*y
+    end
+    if ix < 0x40180000 # 0.84375 <= |x| < 6
+        y = 1 - erfc2(ix,x)
+    else
+        y = 1 - 0x1p-1022
+    end
+    return sign == 1 ? -y : y
```

Comment:
Since `sign` can only be either `1` or `0`, `sign != 0` might be a tiny bit faster alternative.